script.js

```diff
@@ -1,62 +1,55 @@
-const imgArr=["./num0.jpg", "./num1.png", "./num2.png", "./num3.png", "./num4.png", "./num5.png", "./num6.jpeg"
-, "./num7.png", "./num8.png", "./num9.jpg", "./num10.jpg", "./num11.png", "./num12.png", "./num13.png"
-, "./num14.png", "./num15.png", "./num16.jpeg", "./num17.png", "./num18.png", "./num19.jpg"];
-var clickedNumArr=[]; 
-var clickedIdArr=[];
-var randImgArr=[ "./num0.jpg","./num1.png", "./num2.png", "./num3.png", "./num4.png", "./num5.png", "./num6.jpeg"
-, "./num7.png", "./num8.png", "./num9.jpg","./num10.jpg", "./num11.png", "./num12.png", "./num13.png"
-, "./num14.png", "./num15.png", "./num16.jpeg", "./num17.png", "./num18.png", "./num19.jpg"];
+const imgArr = Array(20).fill().map((_, i) => `./num${i}.png`);
+
+let clickedNumArr=[];
+let clickedIdArr=[];
+
+let randImgArr = imgArr.slice();
 
 // Randomize img locations 
 function shuffle(o) {
-  for(var j, x, i = o.length; i; j = parseInt(Math.random() * i), x = o[--i], o[i] = o[j], o[j] = x);
-  return o;
+  return o.sort(() => (Math.random() > 0.5) ? 1 : -1);

```

Comment:
This shuffle function looks really unintuitive. If it works, then you can leave it as is.
I left a much more intuitive implementation here. Code needs to be equally as expressive and legible as it is functional.
Keep in mind that, from the [Array.sort documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort), the array sorted by `.sort()` is sorted [in-place](https://en.wikipedia.org/wiki/In-place_algorithm).

It's important to study the bias of any shuffle algorithm you implement. A simple algorithm to implement with very little bias is the [Fisher-Yates Shuffle](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle), which is also really intuitive. You can try implementing that yourself or finding a ready-made implementation on stackoverflow (by googling `how to shuffle javascript array`).

Reply:
I totally get what happened here. but even after readying about sort() over and over again, I still don't get how passing 1 or -1 into sort shall increase or decrease the index of the array.

Reply:
![image](https://user-images.githubusercontent.com/39483620/174445460-5db54127-b8ca-48e5-98f6-0bb2f41823a7.png)

From the documentation, pretty straightforward.

Reply:
`Math.random() > 0.5` is a coin flip.

The `.sort()` function can take a custom compare function, since sometimes, we need to sort things that are complex objects based on complex rules, so we can't just say `carA > carB` which is what the sort function would by default if not supplied a sort function.

So basically, for each comparison this has to make during the sort, this function flips a coin to decide whether the two elements being compared should be sorted `a after b` or `b after a`.

Feel free to study bubble sort, merge sort, selection sort, and insertion sort if you do not know why the `.sort()` function needs to compare numbers internally. Obviously, the `.sort()` function of JS is none of the 4 I just mentioned, but those 4 are easy enough to be studied in a single day, and give you a pretty great understanding of how sort functions are built.